app/controllers/devise/devise_authy_controller.rb

```diff
@@ -183,6 +175,41 @@ def request_sms
 
   private
 
+  def registration_token_valid?(mfa_config)
+    totp_registration_valid?(mfa_config) || sms_token_valid?(mfa_config)
+  end
+
+  def totp_registration_valid?(mfa_config)
+    begin
+      status = @verify_client.validate_totp_registration(
+        mfa_config.verify_identity, mfa_config.verify_factor_id, params[:token]
+      )
+    rescue StandardError => e
+      # 60306 means the input token is too long.
+      raise e unless e.message.include?('60306')
```

Comment:
why wouldn't we want to raise an error in the case that the input is too long?

Reply:
rather... this seems like specific messaging that might actually be helpful to the user instead of a generic "invalid"?

Reply:
That one, I'm really unsure on. I think @djohnston32 wrote this so there's a change he'd have some background.